src/hotspot/share/memory/memoryReserver.cpp

```diff
@@ -437,7 +437,7 @@ ReservedSpace HeapReserver::Instance::try_reserve_range(char *highest_start,
 
   // Try attach points from top to bottom.
   for (char* attach_point = highest_start;
-       attach_point >= lowest_start && attach_point <= highest_start;  // Avoid wrap around.
+       attach_point >= lowest_start && attach_point <= highest_start && p2u(attach_point) > stepsize;  // Avoid wrap around.
```

Comment:
The `attach_point <= highest_start` part of the test should be removed.
That's the old and incorrect attempt to avoid wrap around.

Reply:
The proposed change is not correct.  Consider if attach_point, lowest_start,
and stepsize are all equal.  In such a case this would stop without trying
that case, even though it should try it.  Changing the comparison with
stepsize to >= would fix that.  But it still wouldn't be correct.  Consider if
attach_point == lowest_start and stepsize > lowest_start.  That should also be
tried.

Reply:
Sorry, I need more investigate.

Reply:
1. The `attach_point <= highest_start` has been removed.
2. Add `attach_point <= stepsize` check at the end of for loop, to make sanitizer silent and avoid warp around. The added check seems a bit deliberately but useful.